### PR TITLE
Send extra debugging information

### PR DESCRIPTION
**Pull request by K-Mistele**
Update documentation, send normal stack trace instead of JSON, add abillity to send extra debugging information

### CODE REVIEW

File: codelighthouse/CodeLighthouse.py [hunk 1]
```diff
@@ -68,20 +68,23 @@ def error(self, exception, email=None, args=None, kwargs=None):
         :return: The error guid if it was successful -- otherwise, returns None
         """
         arguments = CodeLighthouse.format_arguments(args, kwargs)
-        stack_trace = CodeLighthouse.format_stack_trace(exception.__traceback__)
+        parsed_stack_trace = CodeLighthouse.format_stack_trace(exception.__traceback__)
+        stack_trace = ''.join(traceback.format_exception(None, exception, exception.__traceback__))
+        print(stack_trace)
```
Comment (willtraweek) [inline codelighthouse/CodeLighthouse.py]: I don't think we should be printing the stack trace here, should we?

File: docs/python/py_sdk_docs.md [hunk 2]
```diff
@@ -125,6 +133,25 @@ def some_function():
...
+    try:
+        call_a_broken_function()
+    except NameError as e:
+        lighthouse.error(e, email="bob@codelighthouse.io", data=some_data)
+```
+
+Make sure that the data you're passing (via the `data` argument as show above) can be serialized into JSON. Dictionaries
+and native types will easily work, but if you choose to pass an object, pass the class's `__dict__` property instead 
+and ensure that it does not contain circular references. For more information on the `__dict__` property of Python
+classes, refer to [the python docs](https://docs.python.org/3/library/stdtypes.html#object.__dict__).
```
Comment (willtraweek) [inline docs/python/py_sdk_docs.md]: given this info, should we automatically do this for them?

I.E. should we check to see if it works as a serializable object, if not, then we can check to see if it's an object (by trying to pass `object.__dict__` -- if that fails, then we just don't pass it, but we leave a message letting them know that it was non-serializable.

**Review by willtraweek — CHANGES_REQUESTED** [review]

File: docs/python/py_sdk_docs.md (hunk 2 above)
Comment (K-Mistele) [reply docs/python/py_sdk_docs.md]: yeah you're right we should handle this for them - I'll work on it

File: codelighthouse/CodeLighthouse.py (hunk 1 above)
Comment (K-Mistele) [reply codelighthouse/CodeLighthouse.py]: lol yeah that'd be my fault

File: docs/python/py_sdk_docs.md (hunk 2 above)
Comment (K-Mistele) [reply docs/python/py_sdk_docs.md]: 8df4d0dfcd45b7afe67980e8b644b3067df760d8

**Comment by K-Mistele** [comment]
@willtraweek back in your court

File: codelighthouse/CodeLighthouseWebHandler.py [hunk 3]
```diff
@@ -19,6 +19,25 @@ def send_error(self, *args, **kwargs):
...
+
+            # CHECK IF IT HAS AN __DICT__ ATTRIBUTE AND IF SO USE THAT
+            if hasattr(kwargs['user_data'], '__dict__'):
+                kwargs['user_data'] = kwargs['user_data'].__dict__
+            else:
+                kwargs['user_data'] = {'error': 'The passed data was not JSON-serializable'}
+                print('CODELIGHTHOUSE: additional data specified in CodeLighthouse.error() could not be serialized'
+                      'to JSON. Omitting it from the report.')
+
+
```
Comment (willtraweek) [inline codelighthouse/CodeLighthouseWebHandler.py]: You'll need another `json.dumps` in between lines 32 and 33

**Review by willtraweek — CHANGES_REQUESTED** [review]

File: codelighthouse/CodeLighthouseWebHandler.py (hunk 3 above)
Comment (K-Mistele) [reply codelighthouse/CodeLighthouseWebHandler.py]: 55877a5cf3d1705b82e81f3c778cfe67249bfca4 @willtraweek

**Review by willtraweek — APPROVED** [review]

**Pull request merged**